### ISSUE TITLE
>/dev/null | grep will always fail

### ISSUE THREAD

**Issue by austin987** [koalaman/shellcheck#1934]
#### For new checks and feature suggestions
- [X] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh

echo >/dev/null | grep -v foo
```

#### Here's what shellcheck currently says:
```
$ shellcheck myscript
No issues detected!
```

#### Here's what I wanted or expected to see:
```
$ shellcheck myscript
^-- SC20XX: Redirecting stdout then attempting to grep it will always fail
```

Noticed this in upstream ganeti's doc scripts [1] when looking at an upstream issue for failing to build in gentoo [2]. If stdout is redirected to /dev/null (or really, anywhere), then grep gets no input to search, and always fails:
```
austin@laptop ~ $ echo > /dev/null | grep ; echo $?
Usage: grep [OPTION]... PATTERNS [FILE]...
Try 'grep --help' for more information.
2
austin@laptop ~ $ echo > /dev/null | grep '' ; echo $?
1
austin@laptop ~ $ echo > /dev/null | grep -v '' ; echo $?
1
austin@laptop ~ $ echo > /dev/null | grep a ; echo $?
1
```

Of course, is grep is searching a file, that works fine, this only applies when stdout is redirected and then piped to grep.
[1] https://github.com/ganeti/ganeti/blob/master/autotools/check-man-warnings#L38
[2] https://bugs.gentoo.org/719846

**Comment by koalaman** [comment]
With 2030b83607a, all of `cmd > file | cmd`, `cmd | cmd < file` and `cmd < input1 < input2`, and `cmd > output1 > output2` will now receive warnings.

I found a good selection of instances. Some simple ones like your example, but also accidentally duplicated redirections, bad line breaks, `>`s and `&>`s that should have been `2>`s or `<`s. 

Thanks!

**Comment by austin987** [comment]
Great, thanks for the quick turnaround!
